### PR TITLE
Fix corner cases in handling of android lib when compiling against di…

### PR DESCRIPTION
**Pull request by oliviernotteghem**
…rect dependencies only

https://github.com/oliviernotteghem/rules_kotlin/commit/004189ea53792cad74e0b3b8559d4889707cafbb came a slightly outdated branch. Syncing changes with what we're using internally at Uber.

### CODE REVIEW

File: kotlin/internal/jvm/android.bzl [hunk 1]
```diff
@@ -103,7 +95,7 @@ def _kt_android_artifact(
         tags = tags,
         exec_properties = exec_properties,
     )
-    return [base_name, kt_name]
+    return [kt_name, base_name]
```
Comment (nkoroste) [inline kotlin/internal/jvm/android.bzl:98]: why is this needed? that sounds like something is not hermetic

**Comment by Bencodes** [comment]
Going to close this PR since we rolled back to the previous behavior https://github.com/bazelbuild/rules_kotlin/commit/0f4c0af37ea4451fa01e001cf704a8fc6c897d22
